nightfall-optimist/src/event-handlers/transaction-submitted.mjs

```diff
@@ -68,15 +68,20 @@ async function transactionSubmittedEventHandler(eventParams) {
     transaction = await checkAlreadyInBlock(transaction);
     // save transaction if not in block
     if (fromBlockProposer) {
-      saveTransaction({ ...transaction });
+      saveTransaction({ ...transaction }).catch(err =>
```

Comment:
There is already a catch of this error. I don't think this code will behave the same

Reply:
No, the `catch` this instruction is inside only handles synchronous calls, this one is an async call and returns a promise that crashes the app if an error happen, then that is one of the possible ways to handle it.

Reply:
Then, should be remove the `try-catch` block if its not doing anything?  And shouldnt the `checkTransaction` function catch an error as well?

Reply:
We can't remove because `checkAlreadyInBlock` and `checkTransactio` are called with `await` and this `try...catch` handles errors coming from them.